### PR TITLE
Fix play/pause

### LINKED ISSUE #135: Pausing and resuming song seems to be broken.
When selecting the server go-librespot is running on as the device and playing a song, it works perfectly. I then can use the skip buttons both forwards and backwards as well as volume control. I can also tap on the "progress bar" of the song to skip ahead. Pause also works.
However, when trying to resume again, the audio does not start playing. On the spotify app, the Play/Pause button changes accordingly, but the playbackbar doesnt progress and there is no sound.

TLDR:

Steps to replicate:
Play song, pause song and press play again.

Result:
Song does not resume playing, playback bar first freezes, then skips to a random point but playback never actually resumes.

Expected result:
Song starts playing normally from the point it was paused at.

Info:
I am using PulseAudio, as I couldnt get the Bluetooth speaker to work with ALSA (not a go-librespot specific problem, couldnt get it to work with spocon either)

### PR DESCRIPTION
**Pull request by devgianlu**
If the ALSA output loop exists with an EOF or an error the PCM handle will be left dangling. This implies that `Resume()` will not spin up a new output loop and nothing will be actually played.

I don't feel like this is the best fix, but it seems to work. @aykevl Can you have a look, what do you think would be the best approach here?

Fixes #135

### CODE REVIEW

**Comment by aykevl** [comment]
> If the ALSA output loop exists with an EOF

It won't, EOF is handled here:
https://github.com/devgianlu/go-librespot/blob/b20fca8c883a54ed28cdf7f9136003a3088a0df2/output/driver-alsa.go#L297-L299

> or an error the PCM handle will be left dangling.

It should be closed by the `Close` call and set to nil. Though there is a possible small race condition in that case.

I don't understand what the bug is, and therefore how this fixes the bug?

(Also, this PR changes the `pcmHandle` value without holding the lock, so that's going to be racy).

**Comment by aykevl** [comment]
Reading the code I can't think of a case where `pcmHandle` would be set while the ALSA output is paused or closed?
Are you sure this is really happening?

**Comment by devgianlu** [comment]
> It won't, EOF is handled here:

Bad wording, what I meant to say is that the loop exists if an EOF occurs.

> I don't understand what the bug is, and therefore how this fixes the bug?

Additional context is https://github.com/devgianlu/go-librespot/commit/08e546e4953b29afdc5b2753f37e79748db4aa17 where I tried to fix the fact that when starting autoplay by letting the last song of a playlist end there will be no sound. This was traced back to the ALSA output loop exiting because of the EOF (there is no gapless when switching to autoplay). The attempt to fix (in the commit above) was to always report to the player loop that the output loop exited (whatever the reason is), but it seems to me now that this should be internal behaviour of the audio backend. Additionally, that commit broke play/pause because when resuming it is assumed that an output device is present.

TL;DR if the output loop exists because of EOF from the audio source, play/pause breaks. You can actually reproduce it very easily on the current master.

**Comment by aykevl** [comment]
Ah I see. That's actually a bug/misfeature, possibly in my recent refactoring PR. Basically, if you look at the ALSA output, you can see that it will unconditionally close the output when `outputLoop` exits, which happens on every pause. I don't think it's supposed to do that: when calling `Pause()` it should actually pause and not secretly close the output entirely.

See #138 for another way to fix this. The difference is that it keeps the output locked, which should avoid race conditions.

(Incidentally the PulseAudio output works fine on current master).
